overlord/devicestate/devicestate_test.go

```diff
@@ -1132,6 +1135,107 @@ func (s *deviceMgrSuite) TestDeviceManagerEmptySystemModeRun(c *C) {
 	c.Check(s.mgr.SystemMode(), Equals, "run")
 }
 
+const (
+	mountRunMntUbuntuSaveFmt = `26 27 8:3 / %s/run/mnt/ubuntu-save rw,relatime shared:7 - ext4 /dev/fakedevice0p1 rw,data=ordered`
+	mountSnapSaveFmt         = `26 27 8:3 / %s/var/lib/snapd/save rw,relatime shared:7 - ext4 /dev/fakedevice0p1 rw,data=ordered`
+)
+
+func (s *deviceMgrSuite) TestDeviceManagerStartupUC20UbuntuSaveFullHappy(c *C) {
+	modeEnv := &boot.Modeenv{Mode: "run"}
+	err := modeEnv.WriteTo("")
+	c.Assert(err, IsNil)
+	mgr, err := devicestate.Manager(s.state, s.hookMgr, s.o.TaskRunner(), s.newStore)
```

Comment:
this creates a new `mgr` locally, but below `s.mgr` is used and then mgr is used? is it intentional to have different mgrs for the test?

Reply:
The reason is that the modeenv is only read when constructing a manager which gets created in SetUpTest(), but we mock it inside the test again.